lib/src/config/config.rs

```diff
@@ -550,6 +550,64 @@ impl Config {
         parse!(self, name, value, as_float, "a float")
     }
 
+    /// Attempts to retrieve the extra named `name` as an array.
+    ///
+    /// # Errors
+    ///
+    /// If an extra with `name` doesn't exist, returns an `Err` of `NotFound`.
+    /// If an extra with `name` _does_ exist but is not an array, returns a
+    /// `BadType` error.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use std::collections::BTreeMap;
+    /// use rocket::config::{Config, Environment};
+    ///
+    /// let mut table = BTreeMap::new();
+    /// table.insert("my_value".to_string(), 1);
+    ///
+    /// let my_array = vec![table];
+    ///
+    /// let config = Config::build(Environment::Staging)
+    ///     .extra("my_array", my_array)
```

Comment:
Can we make `my_array` a simpler vector? Maybe just `vec![1, 2, 3]` or something?